packages/cozy-intent/src/view/components/WebviewIntentProvider.spec.tsx

```diff
@@ -59,4 +71,50 @@ describe('WebviewIntentProvider', () => {
     expect(await screen.findByText('Hello')).toBeDefined()
     expect(mockChildHandshake).not.toHaveBeenCalled()
   })
+
+  it('does not try to set cozy-bar context if cozy-bar does not exist when provided with a context', async () => {
+    global.cozy = undefined
+
+    render(
+      <WebviewIntentProvider webviewService={mockWebviewService}>
+        Hello
+      </WebviewIntentProvider>
+    )
+
+    expect(await screen.findByText('Hello')).toBeTruthy()
+    expect(mockSetWebviewContext).not.toHaveBeenCalled()
+  })
+
+  it('does not try to set cozy-bar context if cozy-bar does not exist when not provided with a context', async () => {
+    global.cozy = undefined
+
+    render(<WebviewIntentProvider>Hello</WebviewIntentProvider>)
+
+    expect(await screen.findByText('Hello')).toBeTruthy()
+    expect(mockSetWebviewContext).not.toHaveBeenCalled()
+  })
+
+  it('sets cozy-bar context if cozy-bar does exist when provided with a context', async () => {
+    global.cozy = mockCozyBar
+
+    render(
+      <WebviewIntentProvider webviewService={mockWebviewService}>
+        Hello
+      </WebviewIntentProvider>
+    )
+
+    expect(await screen.findByText('Hello')).toBeTruthy()
+    expect(mockSetWebviewContext).toHaveBeenCalledWith(mockWebviewService)
+    expect(mockSetWebviewContext).toBeCalledTimes(1)
+  })
+
+  it('sets cozy-bar context if cozy-bar does exist when not provided with a context', async () => {
+    global.cozy = mockCozyBar
+
+    render(<WebviewIntentProvider>Hello</WebviewIntentProvider>)
+
+    expect(await screen.findByText('Hello')).toBeTruthy()
+    expect(mockSetWebviewContext).toBeCalledWith(expect.any(WebviewService))
+    expect(mockSetWebviewContext).toBeCalledTimes(1)
```

Comment:
in my opinion, there should be 2 tests:
```
it('sets cozy-bar context if cozy-bar does exist when not provided with a context', async () => {
    global.cozy = mockCozyBar

    const { queryByText } = await render(<WebviewIntentProvider>Hello</WebviewIntentProvider>)

    expect(queryByText('Hello')).toBeTruthy()
  })

  it('sets cozy-bar context if cozy-bar does exist when not provided with a context 2', async () => {
    global.cozy = mockCozyBar

    render(<WebviewIntentProvider>Hello</WebviewIntentProvider>)

    await waitFor(() => {
      expect(mockSetWebviewContext).toBeCalledWith(expect.any(WebviewService))
      expect(mockSetWebviewContext).toBeCalledTimes(1)
    })
  })
```

Reply:
Kinda disagree here, the "hello" query is a smoke test in every test to ensure the component doesn't crash. I think it's a bit redundant to have a test for it everytime (it would double the amount of tests)